provisioner/secret.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+var base = require('taskcluster-base');
+var assert = require('assert');
+var debug = require('debug')('aws-provisioner:Secret')
+var taskcluster = require('taskcluster-client');
+
+var Secret = base.Entity.configure({
+  version: 1,
+
+  // TODO: verify these are valid azure keys - heeeelp please Jonas
+  partitionKey: base.Entity.keys.StringKey('token'),
+  rowKey: base.Entity.keys.StringKey('provisionerId'),
+
+  properties: {
+    token: base.Entity.types.String,
+    created: base.Entity.types.Date,
+    provisionerId: base.Entity.types.String,
+    workerType: base.Entity.types.String,
+    secrets: base.Entity.types.JSON,
+  },
+});
+
+// we pass through the json from the HTTP post request body
+Secret.create = function(token, provisionerId, body) {
+  assert(typeof body === 'object');
+  assert(body.workerType);
+  assert(body.secrets);
+  assert(typeof body.secrets === 'object');
+  var properties = {
+    token: token,
+    created: new Date(),
+    provisionerId: provisionerId,
+    workerType: body.workerType,
+    secrets: body.secrets,
+  };
+  debug('Created Secret entity %s: %j', token, body.secrets);
+  return base.Entity.create.call(this, properties);
+};
+
+Secret.prototype.modify = function() {
+  throw new Error('No modifications to secrets are allowed');
+};
+
+Secret.prototype.listAll = function() {
+  throw new Error();
+};
+
```

Comment:
You need a `Secrets.expire` method...
something like this:
```js
Secret.expire = async function(now) {
  assert(now instanceof Date, "now must be given as option");
  var count = 0;
  await base.Entity.scan.call(this, {
    expires: base.Entity.op.lessThan(now)
  }, {
    limit: 250, // max number of concurrent delete operations
    handler: (item) => { count++; return item.remove(true); }
  });
  return count;
};
```

Note, `count` is nice to have for testing... Then you just create a secret and run this with `now = now() + 5 days` and count that at least something was deleted.